### DISCUSSION TITLE
Open child windows on the same workspace as parent windows

### DISCUSSION THREAD

**Discussion by germaniii** [qtile/qtile#2950] [Ideas] [answered]
# Description
Whenever there is an event where a child window is opened, it would open inside the current active workspace instead of the same workspace where the parent window is located. 
One example of this is:
-> I open steam, which usually opens a few windows like the promos and the chat box

-> When I switch workspaces to open another program, the chat box/promos window would open on the current workspace I am in instead of the same workspace as the main steam client.

Idk if that is just how window managers work, but I think it would help in improving workflows.

# Qtile version
Qtile version 0.18.1-3
installed through pacman

**Comment by m-col** [comment, marked as answer]
This is supported following the X11 spec, so it is up to the programs to provide us with the information to do so. Clients should set the `WM_TRANSIENT_FOR` hint on child windows to the window ID of their parent window. For example, a `xprop` on the file open dialog opened by Firefox contains the line `WM_TRANSIENT_FOR(WINDOW): window id # 0x3c0002c`, which points to the main firefox window and lets us stick it on the correct group (and also positioned centred over the parent window). A quick test on the steam help dialog suggests to me that they haven't bothered to implement this. Your best bet would be to post an issue on the steam client bug tracker. If you find other programs that do the same thing it's likely the same issue unfortunately.

**Reply by germaniii** [reply]
oh I see thanks for the explanation
